File: HW2/HW2_U107184/leibniz.cpp
#include<iostream> 
#include<iomanip>
#include<math.h>


int main()
{
	int length, n;
	// input the number of cases to be tested
	std::cin >> length;

	for (int i = 0; i < length; i++)
	{
		std::cout << "Case " << i << ":\n";
		std::cin >> n;
		double sum = 0.0;
		for (int j = 0; j < n; j++)
		{
			sum = sum + (pow(-1,j)) / (2.0 * j + 1);
		}
		sum = sum * 4;
		std::cout << "Pi estimated as: ";
		std::cout << std::fixed << std::setprecision(8) << sum << "\n";
	}
	return 0;
}

File: PY1/PYHW2/leibniz.py
length=int(input())
for i in range(length):
     print( "Case ",i,":",sep='')
     n=int(input())
     pi=0
     for j in range(n):
         pi=pi+((-1)**j/(2*j+1))
     pi=pi*4
     print("Pi estimated as: ",'{:.8f}'.format(pi),sep='')


         

